assemblies/evaluating-rag-systems.adoc

```diff
@@ -0,0 +1,103 @@
+
+*Context Recall*:: Measures the recall of the retrieval component by evaluating whether all necessary information for answering the question is present in the retrieved contexts. High recall ensures that your retrieval system is not missing important information.
+
+*Answer Correctness*:: Compares the generated answer with a ground truth reference answer to measure accuracy. This metric is useful when you have labeled evaluation datasets with known correct answers.
+
+*Answer Similarity*:: Measures the semantic similarity between the generated answer and a reference answer, providing a more nuanced assessment than exact string matching.
+
+=== Use cases for Ragas in AI engineering workflows
+
+Ragas enables AI engineers to accomplish the following tasks:
+
+*Automate quality checks*::
+Create reproducible, objective evaluation jobs that can be automatically triggered after every code commit or model update. This establishes quality gates to prevent regressions and ensures that only high-quality RAG configurations are deployed to production.
+
+*Enable evaluation-driven development (EDD)*::
+Use Ragas metrics to guide iterative optimization. For example, test different chunking strategies, embedding models, or retrieval algorithms against a defined benchmark to discover the optimal RAG configuration that maximizes performance metrics like faithfulness while minimizing computational cost.
+
+*Ensure factual consistency and trustworthiness*::
+Measure the reliability of your RAG system by setting thresholds on metrics like faithfulness. This ensures that responses are consistently grounded in source documents, which is critical for enterprise applications where hallucinations or factual errors are unacceptable.
+
+*Achieve production scalability*::
+Leverage the remote provider pattern with {productname-short} pipelines to execute evaluations as distributed jobs. This allows you to run large-scale benchmarks across thousands of data points without blocking development or consuming excessive local resources.
+
+*Compare model and configuration variants*::
+Run comparative evaluations across different models, retrieval strategies, or system configurations to make data-driven decisions about your RAG architecture. For example, compare the impact of different chunk sizes (512 vs 1024 tokens) or different embedding models on retrieval quality metrics.
+
+=== Ragas provider deployment modes
+
+{productname-short} supports two deployment modes for Ragas evaluation:
+
+*Inline provider*::
+Runs Ragas evaluation in the same process as the Llama Stack server. This mode is suitable for development environments and lightweight evaluation workloads where simplicity and quick iteration are priorities. The inline provider offers:
```

Comment:
Use the inline provider for development and rapid prototyping. It offers the following advantages:
* Faster (? since comparing to just one other option) processing 
also not clear what is meant by "with" - change to "of in-memory operations" or "by using in-memory operations" 

Try to make the sub-bullets parallel in structure:

* Fast processing
* Minimal configuration overhead
* Local development and testing enviroment
* Evaluation of small to ...